scheduler-worker/src/test/java/gov/nasa/jpl/aerie/scheduler/worker/services/SchedulingIntegrationTests.java

```diff
@@ -2744,7 +2746,128 @@ export default () => Goal.CoexistenceGoal({
     assertEquals(SerializedValue.of("fromStem"), peelBanana.serializedActivity().getArguments().get("peelDirection"));
   }
 
-    /**
+  @Test
+  void testJustAfter() {
+    /*
+    Start with a plan with B anchored to A
+    Goal: for each B, place a C
+    And make sure that C ends up in the right place
+     */
+    final var activityDuration = Duration.of(1, Duration.HOUR);
+    final var tenMinutes = Duration.of(10, MINUTES);
+    final var results = runScheduler(
+        BANANANATION,
+        Map.of(
+            new ActivityDirectiveId(2L),
+            new ActivityDirective(
+                tenMinutes,
+                "GrowBanana",
+                Map.of(
+                    "quantity", SerializedValue.of(1),
+                    "growingDuration", SerializedValue.of(activityDuration.in(Duration.MICROSECONDS))),
+                new ActivityDirectiveId(1L),
+                true)),
```

Comment:
Where is the `ActivityDirective` with an ID of `1L` defined? And if it isn't, then this test should be failing, since simulation can't simulate this activity. This applies to `testJustBefore` as well.

Reply:
Nowhere. It simulates because the activity is not found in the greatly named `planActDirectiveIdToSimulationActivityDirectiveId` and thus its anchor id is null. I'll remove this anchor, it makes no sense here anyway.

Reply:
Does this meant that the scheduler alters the directives it is given? Or is removing the anchor on this directive a test-only oddity?

Reply:
no it does not. When looking for the corresponding activity anchor in simulation (in the map), it does not find it (as it does not exist).

Reply:
Alright, I've traced the bug down to L335 of the facade. If the anchor activity does not exist in the map, we ought to throw, as the plan data is scuffed.

Reply:
Added this bit 
```java
if(activity.anchorId()!= null && !planActDirectiveIdToSimulationActivityDirectiveId.containsKey(activity.anchorId())){
      throw new RuntimeException("Activity with id "+ activity.anchorId() + " referenced as an anchor by activity " + activity.toString() + " is not present in the plan");
}
```